indico/modules/events/timetable/controllers/legacy.py

```diff
@@ -133,3 +137,56 @@ def _process(self):
                 entries.append(schedule_contribution(contribution, start_dt=start_dt))
         return jsonify(entries=[serialize_entry_update(x) for x in entries])
+
+
+class RHLegacyTimetableEntryMove(RHManageTimetableEntryBase):
+    """Moves a TimetableEntry into a Session or top-level timetable"""
+
+    def _process_POST(self):
+        current_day = dateutil.parser.parse(request.form.get('day')).date()
+        return jsonify_template('events/timetable/move_timetable_entry.html', event=self.event_new,
+                                top_level_entries=self._get_session_timetable_entries(),
+                                current_day=current_day, timetable_entry=self.timetable_entry,
+                                parent_entry=self.timetable_entry.parent)
+
+    def _process_PATCH(self):
+        self.serializer = TimetableSerializer(True)
+        with track_time_changes():
+            entry_data = self._move_entry(request.json)
+        rv = dict(serialize_entry_update(self.timetable_entry), **entry_data)
+        return jsonify_data(**rv)
+
+    def _move_entry(self, data):
+        rv = {}
+        if data.get('parent_id'):
+            parent_timetable_entry = self.event_new.timetable_entries.filter_by(id=data['parent_id']).one()
+            rv['old'] = self.serializer.serialize_timetable_entry(self.timetable_entry)
+            move_timetable_entry(self.timetable_entry, parent=parent_timetable_entry)
+            rv['session'] = rv['slotEntry'] = self.serializer.serialize_session_block_entry(parent_timetable_entry)
+        elif data.get('day'):
+            new_date = as_utc(dateutil.parser.parse(data['day']))
+            rv['old'] = self.serializer.serialize_timetable_entry(self.timetable_entry)
+            move_timetable_entry(self.timetable_entry, day=new_date)
+        return rv
+
+    def _get_session_timetable_entries(self):
+        entries = {}
+        for day in iterdays(self.event_new.start_dt, self.event_new.end_dt):
+            entries[day.date()] = get_top_timetable_session_blocks(self.event_new, day)
+        return entries
+
+
+class RHLegacyChangeTimetableEntryStartEndDate(RHManageTimetableEntryBase):
+    """Changes the start_dt of a `TimetableEntry`"""
+
+    def _process(self):
+        new_start_dt = as_utc(dateutil.parser.parse(request.form.get('startDate')))
+        new_end_dt = as_utc(dateutil.parser.parse(request.form.get('endDate')))
+        new_duration = new_end_dt - new_start_dt
+        is_session_block = self.timetable_entry.type == TimetableEntryType.SESSION_BLOCK
+        with track_time_changes():
+            if is_session_block:
+                self.timetable_entry.move(new_start_dt)
+            update_timetable_entry_object(self.timetable_entry, {'duration': new_duration})
+            update_timetable_entry(self.timetable_entry, {'start_dt': new_start_dt.astimezone(self.event_new.tzinfo)})
+        return jsonify_data(**serialize_entry_update(self.timetable_entry))
```

Comment:
Add the serialized update on a top-level entry like in the rest of the RHs. You are still modifying the JS instead of complying what it expects.

```python
return jsonify_data(entry=serialize_entry_update(entry), flash=False)
```